docs/blog/2019-03-27-security-concerns-unique-to-static-sites/index.md

```diff
@@ -0,0 +1,128 @@
+
+**TL;DR:** Authenticate users using JSON Web Tokens(JWTs) and dynamically render pages only to authorized users. Any API calls that need to be made in order to fetch content can use the user's JWT and be verified by the API. If you need to access a third party API, any API Keys can be stored securely by your API which is securely authenticated to using the JWT.
+
+This is pretty much the base of the "JAMstack" and allows for private content in client-side sites. Requiring user authentication will cover all use cases involving private data and will also help in situations where data is read publicly but written by authenticated users (e.g. CMS backend or comments section).
+
+### Public Dynamic Content
+
+This should be easy right? It is public content after all. But, if you need anonymous users to be able to both read and write this data, you should be concerned. Content read publicly, but written privately was covered by previous section. Admittedly, public reading and writing of content is less common.
+
+Medium's "clap" feature, up-voting on Hacker News or Reddit, liking something on Facebook or Twitter, these all require authentication. Even public APIs like GitHub's requite authentication or sites like wikipedia. Even if anyone in the world can create an account and access these tools.
+
+Why address this scenario? Two reasons:
+
+- **Simplicity** - Consider a small blog or non-profit site where authentication is not already needed. To develop it would require a lot of time and long term investment. For users authenticating to small blog just to hit "like" is bad user experience and is probably a hurtle they will not bother jumping.
+
+- **Privacy** - No matter the size of your platform, requiring user to authenticate means having to secure private data. This requires time and money. Alternatively, as a platform, you may decide your users' privacy is more important.
+
+Using Medium's "clap" feature as example, what are the issues? First, we need some kind of database to store the claps. This is easy enough, but do we leave no security mechanism on this database for reads and writes? We could add security, but then the API key we need has to be shipped in the code and the user can access it. Either way, your more nefarious users could access this database; reading and (more importantly) writing to their devious heart's content.
+
+There is a small blog that does this very thing. They build the Firebase credentials into the code at build time. Because this is a small blog, and their is no real incentive for someone to add or delete claps, this blogger is probably safe. What is the worse case scenario? Someone deletes their claps? Gives all their posts a million claps? Maybe this is defeating the point of a security post. But security is bout threat assessment. This blogger is clearly not worried about a nation state hacking their claps or someone gaming a system. Nothing but authentication is going to solve those kinds of issues.
```

Comment:
```suggestion
There is a small blog that does this very thing. They build the Firebase credentials into the code at build time. Because this is a small blog, and their is no real incentive for someone to add or delete claps, this blogger is probably safe. What is the worse case scenario? Someone deletes their claps? Gives all their posts a million claps? Maybe this is defeating the point of a security post. But security is about threat assessment. This blogger is clearly not worried about a nation state hacking their claps or someone gaming the system. Nothing but authentication is going to solve those kinds of issues.
```